test/default_field_linter.lean

```diff
@@ -0,0 +1,66 @@
+import Mathlib.Algebra.Ring.Defs
+
+def Foo := Unit
+
+
+/--
+error: While evaluating default nsmul := nsmulRec:
+  failed to synthesize instance
+    Zero Foo
+---
+error: While evaluating default zsmul := zsmulRec:
+  failed to synthesize instance
+    Zero Foo
+-/
+#guard_msgs in
+instance : Ring Foo where
+  add_assoc _ _ _ := rfl
+  zero := ()
+  zero_add _ := rfl
+  add_zero _ := rfl
+  add_comm _ _ := rfl
+  mul _ _ := ()
+  left_distrib _ _ _ := rfl
+  right_distrib _ _ _ := rfl
+  zero_mul _ := rfl
+  mul_zero _ := rfl
+  mul_assoc _ _ _:= rfl
+  one := ()
+  one_mul _ := rfl
+  mul_one _ := rfl
+  add_left_neg _ := rfl
+
+-- this is the fix to the errors above
+instance : Zero Foo where
+  zero := ()
+
+instance : Add Foo where
+  add _ _ := ()
+
+instance : Neg Foo where
+  neg _ := ()
+
+/--
+warning: Using default value nsmul := nsmulRec, which may cause typeclass diamonds for `Module ℕ _`.
+If you are sure this is not an issue, write nsmul := nsmulRec explicitly [linter.structureDiamondDefaults]
+---
+warning: Using default value zsmul := zsmulRec, which may cause typeclass diamonds for `Module ℤ _`.
+If you are sure this is not an issue, write zsmul := zsmulRec explicitly [linter.structureDiamondDefaults]
+-/
```

Comment:
I don't think this is really helpful for beginners. Of course they are not sure that something they don't understand is not an issue.

Reply:
I guess this could be toned down; really the point is to make `nsmul := nsmulRec` appear in source so that a reviewer can pick on it.

Reply:
Edited.